bindings/terra/bishop_c.h

```diff
@@ -16,7 +16,8 @@
 #ifndef __BISHOP_C_H__
 #define __BISHOP_C_H__
 
-#include "legion_c.h"
+#define LEGION_ENABLE_C_BINDINGS
+#include "legion.h"
```

Comment:
My first concern with this approach is that if someone does:

```
#include "legion.h"
#include "bishop_c.h"
```

Then on the first include of `legion.h` the define won't be set, and then the include will be a nop on the second include. Is `leigon.h` set up to mitigate this?

My second concern is that since this is called `bishop_c.h` it is presumably a pure-C file. We should check with @magnatelee to be sure, but it really shouldn't be named `*_c.h` if it isn't. If so, then it would be really really bad to include any C++ functionality. It should really just include the C API directly.